hazelcast-client/src/test/java/com/hazelcast/client/cache/nearcache/ClientNearCacheInvalidationTest.java

```diff
@@ -0,0 +1,344 @@
+    public static Collection<Object[]> parameters() {
+        return Arrays.asList(new Object[][]{
+                {false, InMemoryFormat.BINARY},
+                {false, InMemoryFormat.OBJECT},
+                {true, InMemoryFormat.BINARY},
+                {true, InMemoryFormat.OBJECT},
+        });
+    }
+
+    private HazelcastInstance member;
+    private TestHazelcastFactory hazelcastFactory;
+    private NearCacheTestContext testContext;
+
+    @Before
+    public void setup() {
+        hazelcastFactory = new TestHazelcastFactory();
+        member = hazelcastFactory.newHazelcastInstance(getConfig());
+        ClientConfig clientConfig = createClientConfig();
+        clientConfig.addNearCacheConfig(createNearCacheConfig(inMemoryFormat));
+        HazelcastClientProxy client = (HazelcastClientProxy) hazelcastFactory.newHazelcastClient(clientConfig);
+        NearCacheManager nearCacheManager = client.client.getNearCacheManager();
+        CachingProvider provider = HazelcastClientCachingProvider.createCachingProvider(client);
+        HazelcastClientCacheManager cacheManager = (HazelcastClientCacheManager) provider.getCacheManager();
+        HazelcastServerCacheManager memberCacheManager = (HazelcastServerCacheManager) HazelcastServerCachingProvider
+                .createCachingProvider(member).getCacheManager();
+
+        ICache<Object, String> cache = cacheManager.createCache(DEFAULT_CACHE_NAME, createCacheConfig(inMemoryFormat));
+        ICache<Object, String> memberCache = member.getCacheManager().getCache(
+                CacheUtil.getPrefixedCacheName(DEFAULT_CACHE_NAME, null, null));
+
+        NearCache<Data, String> nearCache = nearCacheManager.getNearCache(
+                cacheManager.getCacheNameWithPrefix(DEFAULT_CACHE_NAME));
+
+        testContext = new NearCacheTestContext(client, member, cacheManager, memberCacheManager, nearCacheManager, cache,
+                memberCache, nearCache);
+
+        // add 1 K-V pair in the cache
+        testContext.cache.put("a", "b");
+        testContext.cache.get("a");
+
+    }
+
+    @After
+    public void tearDown() {
+        hazelcastFactory.shutdownAll();
+    }
+
+    @Test
+    public void when_cacheDestroyed_invalidationEventIsReceived() {
+        final AtomicInteger counter = new AtomicInteger();
+        registerInvalidationListener(counter);
+        destroy();
+        assertInvalidationEventReceivedExactlyOnce(counter);
+    }
+
+    @Test
+    public void when_cacheCleared_invalidationEventIsReceived() {
+        final AtomicInteger counter = new AtomicInteger();
+        registerInvalidationListener(counter);
+        clear();
+        assertInvalidationEventReceivedExactlyOnce(counter);
+    }
+
+    @Test
+    public void when_cacheClosed_invalidationEventIsNotReceived() {
+        final AtomicInteger counter = new AtomicInteger();
+        registerInvalidationListener(counter);
+        close();
+        assertTrueAllTheTime(new AssertTask() {
```

Comment:
I think this code is a good candidate for a `assertInvalidationEventNeverReceived()` method, since we have it three times in the test.